### PR TITLE
make repository dispatch call to run aci-examples tests

### PR DESCRIPTION
**Pull request by hgarvison**
Make repository dispatch call to aci-examples repo to run tests for sidecars automatically.

Review in conjunction with: https://github.com/microsoft/confidential-aci-examples/pull/116

### CODE REVIEW

File: .github/workflows/push_attestation_image.yml [hunk 1]
```diff
@@ -62,3 +62,15 @@ jobs:
           push: true
           tags: |
             ${{ secrets.REGISTRY_NAME }}.${{ secrets.REGISTRY_DOMAIN }}/attestation:${{ steps.get_image_tag.outputs.image_tag }}
+
+      - name: Run Attestation Tests
+        env:
+          PR_NUMBER: ${{ github.event.number }}
+        run: |
+          curl -L \
```
Comment (ksayid) [inline .github/workflows/push_attestation_image.yml]: should a `--fail` or `--fail-with-body` flag be added to the `curl` command? I think `curl` will treat an HTTP response with a 4xx or 5xx status code as successful. Adding the flags would make `curl` return a non-zero exit code. 

I don't think we'd want the workflow to proceed if the tests haven't been started.

File: .github/workflows/push_encfs_image.yml [hunk 2]
```diff
@@ -57,8 +58,24 @@ jobs:
...
             ${{ secrets.REGISTRY_DOMAIN }} \
-            encfs:$image_tag \
+            encfs:${{ steps.get_image_tag.outputs.image_tag }} \
             --skip-login
+
+      - name: Run ENCFS Tests
+        env:
+          PR_NUMBER: ${{ github.event.number }}
+        run: |
+          curl -L \
```
Comment (ksayid) [inline .github/workflows/push_encfs_image.yml]: same here?

File: .github/workflows/push_skr_image.yml [hunk 3]
```diff
@@ -56,8 +57,24 @@ jobs:
...
             ${{ secrets.REGISTRY_DOMAIN }} \
-            skr:$image_tag \
+            skr:${{ steps.get_image_tag.outputs.image_tag }} \
             --skip-login
+
+      - name: Run SKR Tests
+        env:
+          PR_NUMBER: ${{ github.event.number }}
+        run: |
+          curl -L \
```
Comment (ksayid) [inline .github/workflows/push_skr_image.yml]: and here?

File: .github/workflows/push_attestation_image.yml (hunk 1 above)
Comment (hgarvison) [reply .github/workflows/push_attestation_image.yml]: That works for me. I wasn't aware of that curl flag.

**Review by ksayid — APPROVED** [review]
LGTM

**Pull request closed without merging**